From 7ecbd8e348655d1806e7eca5d1093014eb0d2c03 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Mon, 8 Jun 2020 17:13:55 +0200
Subject: [PATCH 55/77] dsm: fix mismatched pointer

Fixes this warning with clang10:
incompatible pointer types passing 'u_long *' (aka 'unsigned long *') to parameter of type 'uint32_t *' (aka 'unsigned int *')

Similar to what is done in access/smb2.c

(cherry picked from commit 009bce07f3263158e439f8b99cf6d74d782da303)
Signed-off-by: Thomas Guillem <thomas@gllm.fr>
---
 modules/access/dsm/access.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/modules/access/dsm/access.c b/modules/access/dsm/access.c
index 3281507cc0..1c60725de5 100644
--- a/modules/access/dsm/access.c
+++ b/modules/access/dsm/access.c
@@ -302,10 +302,12 @@ static int get_address( stream_t *p_access )
         struct addrinfo *p_info = NULL;
 
         /* Is this a netbios name on this LAN ? */
+        uint32_t ip4_addr;
         if( netbios_ns_resolve( p_sys->p_ns, p_sys->url.psz_host,
                                 NETBIOS_FILESERVER,
-                                &p_sys->addr.s_addr) == 0 )
+                                &ip4_addr) == 0 )
         {
+            p_sys->addr.s_addr = ip4_addr;
             strlcpy( p_sys->netbios_name, p_sys->url.psz_host, 16);
             return VLC_SUCCESS;
         }
-- 
2.32.0 (Apple Git-132)

